src/RepositoryStore.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * This file is part of Factory Muffin.
+ *
+ * (c) Graham Campbell <graham@alt-three.com>
+ * (c) Scott Robertson <scottymeuk@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace League\FactoryMuffin;
+
+use League\FactoryMuffin\Exceptions\DeleteMethodNotFoundException;
+use League\FactoryMuffin\Exceptions\DeletingFailedException;
+use League\FactoryMuffin\Exceptions\MethodNotFoundException;
+use League\FactoryMuffin\Exceptions\SaveMethodNotFoundException;
+
+/**
+ * This is the model store class.
+ *
+ * @author Michael Bodnarchuk <davert@codeception.com>
+ */
+class RepositoryStore extends ModelStore
+{
+    /**
+     * Methods to flush changes to storage.
+     *
+     * @var string
+     */
+    protected $flushMethod = 'flush';
+
+    /**
+     * Instance of a class that performs flushes (EntityManager for Doctrine).
+     *
+     * @var
+     */
+    protected $storage;
+
+    public function __construct($storage)
+    {
+        $this->storage = $storage;
+        $this->saveMethod = 'persist';
+        $this->deleteMethod = 'remove';
+    }
+
+    /**
+     * Save our object to the db, and keep track of it.
+     *
+     * @param object $model The model instance.
+     *
+     * @throws \League\FactoryMuffin\Exceptions\SaveMethodNotFoundException
+     * @throws \League\FactoryMuffin\Exceptions\MethodNotFoundException
+     *
+     * @return mixed
```

Comment:
on other side, I can catch any exception and return false if exception is thrown...